docs/proposals/20241029-sidecarset-inject-serviceaccount.md

```diff
@@ -0,0 +1,127 @@
+authors:
+  - "@magicsong"
+reviewers:
+  - "@furykerry"
+  - "@zmberg"
+creation-date: 2024-10-29
+last-updated: 2024-11-06
+status: 
+
+---
+# ServiceAccount auto injection in SidecarSet
+## Table of Contents
+
+- [ServiceAccount auto injection in SidecarSet](#serviceaccount-auto-injection-in-sidecarset)
+  - [Table of Contents](#table-of-contents)
+  - [Summary](#summary)
+  - [Motivation](#motivation)
+    - [Goals](#goals)
+    - [Non-Goals](#non-goals)
+  - [Proposal](#proposal)
+    - [User Stories](#user-stories)
+    - [Notes/Constraints/Caveats](#notesconstraintscaveats)
+    - [Risks and Mitigations](#risks-and-mitigations)
+  - [Design Details](#design-details)
+    - [API Changes](#api-changes)
+    - [Implementation Strategy](#implementation-strategy)
+  - [Graduation Criteria](#graduation-criteria)
+  - [Drawbacks](#drawbacks)
+  - [Alternatives](#alternatives)
+
+---
+
+## Summary
+
+This KEP proposes the ability to inject a ServiceAccount into a SidecarSet in Kruise. This feature will allow sidecar containers managed by Kruise's SidecarSet to use specific ServiceAccounts, enabling better access control and security configurations.
+
+## Motivation
+
+Currently, sidecars injected by SidecarSet inherit the ServiceAccount of the main application container. In certain use cases, sidecars may require distinct permissions and should be able to specify their own ServiceAccounts. This proposal aims to provide a solution for injecting ServiceAccounts into SidecarSets, giving users finer control over sidecar permissions.
+
+related issue: https://github.com/openkruise/kruise/issues/1747
+
+### Goals
+
+- Allow users to specify a ServiceAccount for sidecar containers managed by SidecarSet.
+- Improve the security and flexibility of permission management for sidecars.
+
+### Non-Goals
+
+- Overhauling the existing ServiceAccount system in Kubernetes.
+- Providing role-based or permission-specific configurations beyond the ServiceAccount level.
+
+## Proposal
+
+### User Stories
+
+- **Story 1**: As a user, I want to assign a dedicated ServiceAccount to a logging sidecar injected by SidecarSet to limit its access to certain resources without affecting the main application container.
+- **Story 2**: As a developer, I want SidecarSet to inject sidecars with different ServiceAccounts depending on the permissions required by each sidecar. A sidecar container needs to access Kubernetes resources such as ConfigMaps, Secrets, or interact with the API server for other reasons, it would require a ServiceAccount with appropriate permissions. Injecting the ServiceAccount directly through the SidecarSet simplifies this process and makes it more manageable.
+
+### Notes/Constraints/Caveats
+
+This feature must ensure compatibility with existing Kruise ServiceAccount management and avoid conflicts when updating ServiceAccounts.
+
+### Risks and Mitigations
+
+- **Risk**: Potential misconfiguration may lead to sidecars running with unintended permissions.
```

Comment:
consider add a serviceaccount white list similar to https://openkruise.io/docs/user-manuals/sidecarset#inject-pod-metadata-annotations

Reply:
I understand that when you mention "whitelist," you are referring to "role," right? In this scenario, it is ultimately the user's responsibility to create the RoleBinding, and Kruise should not manage this Role

Reply:
the whitelist is to control which sidecarset can inject which serviceaccount, so as to control the risk that un-intended user of sidecarset inject serviceaccounts.